src/DicomWebClient/Services/ServiceBase.cs

```diff
@@ -46,16 +46,22 @@ public bool TryConfigureServiceUriPrefix(string uriPrefix)
         {
             Guard.Against.NullOrWhiteSpace(uriPrefix, nameof(uriPrefix));
 
+            if (_httpClient.BaseAddress == null)
+            {
+                throw new InvalidOperationException("BaseAddress is not configured; call ConfigureServiceUris(...) first");
+            }
+
+            Uri newServiceUri = null;
             try
             {
-                var newServiceUri = new Uri(_httpClient.BaseAddress, uriPrefix);
+                newServiceUri = new Uri(_httpClient.BaseAddress, uriPrefix);
                 Guard.Against.MalformUri(newServiceUri, nameof(uriPrefix));
                 RequestServicePrefix = $"{uriPrefix.Trim('/')}/";
                 return true;
             }
             catch (Exception ex)
```

Comment:
what if `ex` is a `StackOverflowException` or a `OutOfMemoryException` or a `ThreadAbortException`?

maybe add `when ((!ex is SystemException))`

Reply:
Will fix in a separate PR.